backend/controllers/iepDetails.js

```diff
@@ -74,8 +74,54 @@ const getIepDetailsFactory = elite2Api => {
     }
   }
 
+  const iconForDifference = {
+    '1': 'Green_arrow.png',
+    '2': 'Double_green_arrow.png',
+    '3': 'TripleGreenArrow.png',
+    '-1': 'Red_arrow.png',
+    '-2': 'Double_red_arrow.png',
+  }
+
+  const levelToIntMap = {
+    Entry: 1,
+    Basic: 2,
+    Standard: 3,
+    Enhanced: 4,
+  }
+
+  const getPossibleLevels = async (context, currentIepLevel, agencyId) => {
+    const levels = await elite2Api.getAgencyIepLevels(context, agencyId)
+
+    return levels
+      .filter(level => level.iepDescription !== currentIepLevel)
+      .filter(level => level.iepDescription !== 'Entry')
+      .map(level => {
+        let diff
+        if (levelToIntMap[level.iepDescription] && levelToIntMap[currentIepLevel]) {
+          diff = levelToIntMap[level.iepDescription] - levelToIntMap[currentIepLevel]
+        } else {
+          // This is a custom level for which we do not have an icon. Always show it last.
+          diff = 1000
+        }
+        return {
+          title: level.iepDescription,
+          value: level.iepLevel,
+          image: iconForDifference[diff.toString()] ? iconForDifference[diff.toString()] : '',
+          diff,
+        }
+      })
+      .sort((a, b) => (Math.abs(a.diff) > Math.abs(b.diff) ? 1 : -1))
```

Comment:
Might not be reading this right, but would have thought the `abs` would mean that if your current level is `basic` then the order of  `entry` or `standard` is  non deterministic as they would both have the same diff? 
(or maybe it is deterministic based on iteration order? - which isn't very explicit)

Reply:
You are right, they will have the same diff, and that is intentional as the diff represents the delta for ordering purposes.

Generally we want the ordering to be 1 step (up or down), 2 steps (up or down), 3 steps (only up possible as Entry cannot be assigned from this page). When it's two 1 step changes we want the delta to be the same for ordering purposes. (I am not sure I am explaining this right...)

Reply:
oh.. maybe that's not a good example as `entry` is filtered out - but maybe if the current level is `standard` and the possible levels are `basic` and `enhanced`

Reply:
Yeah, the above applies to Standard change to Basic and Enhanced - the delta between them is one, and we're ordering by deltas - it doesn't really matter which order those two appear in as their deltas are identical (they appear alphabetically currently). 

The absolute is there to account for two negative steps. For example, current level is Enhanced, the two options are Standard (diff: -1) and Basic (diff: -2). Standard is one delta down, Basic is two deltas down. However, if we compared them by difference `-1 > -2`, so Basic would appear first even though it has the higher delta.

This broke my brain for quite a long time...

Reply:
cool, that makes sense to me.

Wonder if it would be a bit more explicit to order by the magnitude from the `levelToIntMap` instead.

Reply:
That would work for almost all cases, but if the current level is Enhanced (mag. 4), the two options are Basic (2) and Standard (3). Ordering by ascending will show Basic first (delta = 2) and then Standard (delta 1) because Standard's magnitude is bigger.

Ordering in a descending order, when the level is Basic, the two options are Standard (mag. 2) and Enhanced (mag. 3). Ordering by magnitude descending descending would show Enhanced first (delta = 2) and Standard second (delta = 1).

Perhaps I could make the `Make.abs(a.diff)` more explicit by assigning it a variable and calling it delta? As what this is comparing on is not the same thing as the difference - the difference is used for assigning the right icon and negative numbers make sense there whereas we're looking for the delta to be always positive for ordering purposes.

Reply:
Have a delta and put a comment that hopefully makes ordering a bit clearer.